pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -89,6 +89,20 @@ func (ts *TaskSpec) Validate(ctx context.Context) *apis.FieldError {
 	if err := ValidateResourcesVariables(ts.Steps, ts.Resources); err != nil {
 		return err
 	}
+
+	if err := ValidateResults(ts.Results); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidateResults(results []TaskResult) *apis.FieldError {
+	for index, result := range results {
+		if !resultNameFormatRegex.MatchString(result.Name) {
+			return apis.ErrInvalidKeyName(result.Name, fmt.Sprintf("results[%x].name", index), fmt.Sprintf("Name could consist of alphanumeric characters, '-', '_', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my-name',  or 'my_name', regex used for validation is '%s')", ResultNameFormat))
```

Comment:
`results[%x].name` <- this will use base 16 according to https://golang.org/pkg/fmt/#hdr-Printing so result 10 would be printed as `results[a].name` I think? Maybe `%d` would make more sense here?

`Name could consist of` <- suggest `Name must consist of`

Reply:
Done, thanks